File: GodelOS/tests/fixture_loader.js
/**
 * Test fixture utilities for frontend tests
 */

export class FixtureLoader {
  static async loadFixture(fixtureName, subfolder = null) {
    const path = subfolder 
      ? `./fixtures/${subfolder}/${fixtureName}.json`
      : `./fixtures/${fixtureName}.json`;
    
    const response = await fetch(path);
    if (!response.ok) {
      throw new Error(`Fixture not found: ${path}`);
    }
    return response.json();
  }
  
  static async loadCognitiveState(stateName = 'default') {
    return this.loadFixture(stateName, 'cognitive_states');
  }
  
  static async loadApiResponse(endpoint) {
    return this.loadFixture(endpoint, 'api_responses');
  }
  
  static async loadSampleData(dataType) {
    return this.loadFixture(dataType, 'sample_data');
  }
}

File: GodelOS/tests/fixture_loader.py
"""
Test fixture utilities for GödelOS

This module provides utilities for loading test fixtures in a consistent way.
"""

import json
from pathlib import Path
from typing import Any, Dict

FIXTURES_DIR = Path(__file__).parent / "fixtures"

def load_fixture(fixture_name: str, subfolder: str = None) -> Any:
    """Load a test fixture from JSON file"""
    if subfolder:
        fixture_path = FIXTURES_DIR / subfolder / f"{fixture_name}.json"
    else:
        fixture_path = FIXTURES_DIR / f"{fixture_name}.json"
    
    if not fixture_path.exists():
        raise FileNotFoundError(f"Fixture not found: {fixture_path}")
    
    with open(fixture_path) as f:
        return json.load(f)

def load_cognitive_state(state_name: str = "default") -> Dict:
    """Load a cognitive state fixture"""
    return load_fixture(state_name, "cognitive_states")

def load_api_response(endpoint: str) -> Dict:
    """Load an API response fixture"""
    return load_fixture(endpoint, "api_responses")

def load_sample_data(data_type: str) -> Any:
    """Load sample data fixture"""
    return load_fixture(data_type, "sample_data")
